pkg/reconciler/broker/controller.go

```diff
@@ -111,17 +112,17 @@ func NewController(
 	})
 
 	// Reconcile Broker (which transitively reconciles the triggers), when Subscriptions
-	// to those triggers change.
+	// that I own are changed.
 	subscriptionInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
 		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("Broker")),
 		Handler:    controller.HandleAll(impl.EnqueueControllerOf),
 	})
 
-	// Reconcile Broker (which transitively reconciles the triggers), when subscriptions
-	// to those triggers change.
+	// Reconcile trigger (by enqueuing the broker specified in the label) when subscriptions
+	// of triggers change.
 	subscriptionInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
-		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("Broker")),
-		Handler:    controller.HandleAll(impl.EnqueueControllerOf),
+		FilterFunc: pkgreconciler.LabelExistsFilterFunc("eventing.knative.dev/broker"),
```

Comment:
nit: in Serving we would generally hoist these into labels in `pkg/apis/eventing`

Reply:
ack, how about I do that in a followup?